client-app/core/composables/useGoogleAnalytics.ts

```diff
@@ -17,9 +21,46 @@ type CustomEventNamesType = "place_order" | "clear_cart";
 type EventParamsType = Gtag.ControlParams & Gtag.EventParams & Gtag.CustomParams;
 type EventParamsExtendedType = EventParamsType & { item_list_id?: string; item_list_name?: string };
 
-const { storeSettings } = useAppContext();
+const DEBUG_PREFIX = "[GA]";
 
-const isAvailableGtag: Readonly<boolean> = Boolean(IS_CLIENT && storeSettings.googleAnalyticsEnabled && window.gtag);
+const MODULE_KEYS = {
+  ID: "VirtoCommerce.GoogleEcommerceAnalytics",
+  ENABLE_STATE: "GoogleAnalytics4.EnableTracking",
+  TRACK_ID: "GoogleAnalytics4.MeasurementId",
+};
+
+const { isThemeContextReady, modulesSettings } = useThemeContext();
+
+watch(isThemeContextReady, (isReady: boolean) => {
+  if (!canUseDOM) {
+    return;
+  }
+  if (isReady) {
+    const moduleSettings = modulesSettings.value?.find((el) => el.moduleId === MODULE_KEYS.ID);
+    const isGoogleAnalyticsEnabled = !!moduleSettings?.settings?.find((el) => el.name === MODULE_KEYS.ENABLE_STATE)
+      ?.value;
```

Comment:
May be make separate composable which will map it to objects? And then use
```
const { storeSettings } = useThemeContext();
const { modulesSettings } = useModulesSettings(storeSettings);

const value = modulesSettings[MODULE_ID][ENABLE_TRACKING];
...
```

Reply:
will be done with the next module